### PR TITLE
[DOCS] Update dynamic mapping docs to clarify supported match_mapping_type

### LINKED ISSUE #82965: Confusing documentation about dynamic types
In [this section](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-templates.html#match-mapping-type) it says (emphasis mine):

> The match_mapping_type is the data type detected by the JSON parser. Because JSON doesn’t distinguish a long from an integer or a double from a float, it always chooses the wider data type such as **long** for integers and **double** for floating-point numbers.

Later on it says in the table (emphasis mine):

| JSON data type  | `"dynamic":"true"` | `"dynamic":"runtime"` |
| ------------- | ------------- | ------------- |
| `double`  | **`float`**  | `double` |
| `integer`  | **`long`**  | `long` |

This two parts of documentation are contradicting. In the table it is picked `float` and not `double` for floating point numbers. For integers, documentation matches.

### PR DESCRIPTION
**Pull request by lockewritesdocs**
The current dynamic mappings documentation is confusing in relation to `double` and `float` types. This PR adds a note to clarify the distinction, and also updates the data type detected by the JSON parser from `integer` to `long`.

Preview link: https://elasticsearch_83274.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/dynamic-templates.html#match-mapping-type

Closes #82965

### CODE REVIEW

File: docs/reference/mapping/dynamic/templates.asciidoc [hunk 1]
```diff
@@ -123,6 +123,11 @@ JSON doesn't distinguish a `long` from an `integer` or a `double` from
 a `float`, it always chooses the wider data type such as `long` for integers
 and `double` for floating-point numbers.
 
+NOTE: With dynamic mappings, {es} will always choose the wider data type. The
+one exception is `float`, which requires less storage space than `double` and
```
Comment (mitar) [inline docs/reference/mapping/dynamic/templates.asciidoc:127]: This is conflicting with the line above: "always chooses the wider data type such as `long` for integers and `double` for floating-point numbers."

File: docs/reference/mapping/dynamic/field-mapping.asciidoc [hunk 2]
```diff
@@ -21,7 +21,7 @@ h| JSON data type h| `"dynamic":"true"` h| `"dynamic":"runtime"`
  |`null` 2*| No field added
  |`true` or `false` 2*| `boolean`
  |`double` | `float` | `double`
- |`integer` 2*| `long`
+ |`long` 2*| `long`
```
Comment (mitar) [inline docs/reference/mapping/dynamic/field-mapping.asciidoc:24]: I also think clarifying that the first column is JavaScript parser column and the second two columns are ES types would be useful.

File: docs/reference/mapping/dynamic/templates.asciidoc [hunk 3]
```diff
@@ -118,10 +118,12 @@ to map `string` fields as either indexed fields or runtime fields.
 [[match-mapping-type]]
 ==== `match_mapping_type`
 
-The `match_mapping_type` is the data type detected by the JSON parser. Because
-JSON doesn't distinguish a `long` from an `integer` or a `double` from
-a `float`, it always chooses the wider data type such as `long` for integers
-and `double` for floating-point numbers.
```
Comment (lockewritesdocs) [inline docs/reference/mapping/dynamic/templates.asciidoc:124]: @javanna, given the discussions in #82965, I think that this statement is not true.

Comment (javanna) [reply docs/reference/mapping/dynamic/templates.asciidoc:124]: ++

Comment (javanna) [reply docs/reference/mapping/dynamic/templates.asciidoc:124]: I wonder if we should still mention that JSON does not distinguish, and in short, any parsed floating point number is considered a double json data type, while any parsed integer number is considered a long. Up to you.

**Review by javanna — APPROVED** [review]

File: docs/reference/mapping/dynamic/templates.asciidoc (hunk 3 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/dynamic/templates.asciidoc]: Thanks @javanna -- I'll add that clarification.

**Pull request merged**